src/main/java/io/temporal/internal/replay/StartChildWorkflowExecutionParameters.java

```diff
@@ -232,42 +245,46 @@ public ParentClosePolicy getParentClosePolicy() {
     return parentClosePolicy;
   }
 
+  public ChildWorkflowCancellationType getCancellationType() {
+    return cancellationType;
+  }
+
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     StartChildWorkflowExecutionParameters that = (StartChildWorkflowExecutionParameters) o;
     return executionStartToCloseTimeoutSeconds == that.executionStartToCloseTimeoutSeconds
         && taskStartToCloseTimeoutSeconds == that.taskStartToCloseTimeoutSeconds
-        && Objects.equals(namespace, that.namespace)
+        && cancellationType == that.cancellationType
+        && Objects.equal(namespace, that.namespace)
         && Arrays.equals(input, that.input)
-        && Objects.equals(taskList, that.taskList)
-        && Objects.equals(workflowId, that.workflowId)
-        && Objects.equals(workflowType, that.workflowType)
+        && Objects.equal(taskList, that.taskList)
+        && Objects.equal(workflowId, that.workflowId)
+        && Objects.equal(workflowType, that.workflowType)
         && workflowIdReusePolicy == that.workflowIdReusePolicy
-        && Objects.equals(retryParameters, that.retryParameters)
-        && Objects.equals(cronSchedule, that.cronSchedule)
-        && Objects.equals(context, that.context)
-        && Objects.equals(parentClosePolicy, that.parentClosePolicy);
+        && Objects.equal(retryParameters, that.retryParameters)
+        && Objects.equal(cronSchedule, that.cronSchedule)
+        && Objects.equal(context, that.context)
+        && parentClosePolicy == that.parentClosePolicy;
```

Comment:
Not sure if move to reference equality was intended?

Reply:
For enums == is a way to go: https://intellipaat.com/community/1254/comparing-java-enum-members-or-equals

BTW this method is autogenerated by my IDE.